Docs/developer_guide/script_repository/gui.md

```diff
@@ -1223,6 +1223,25 @@ On some systems, *shell=True* must be specified as well.
 
 ### Download and install extension
 
+:::{versionadded} 5.4
+Slicer introduces {func}`slicer.app.installExtensionFromServer()`, which
+simplifies the process of downloading and installing extensions. The updated
+approach is as follows:
+:::
+
+```python
+extensionName = 'SlicerIGT'
+em = slicer.app.extensionsManagerModel()
+em.interactive = False  # prevent display of popups
+restart = True
+if not em.installExtensionFromServer(extensionName, restart):
+  raise ValueError(f"Failed to install {extensionName} extension")
+```
+
+:::{deprecated} 5.4
+The previous method used is now deprecated and no longer recommended.
```

Comment:
What previous method? Can you describe it (specify method name, etc )?

Reply:
```
The approach presented below is now deprecated and no longer recommended.
```

Reply:
Instead of this, I was thinking to create a python package called `slicer-install` (or `slicer-env` or ... ) that would provide a way to:
* install Slicer
* install extensions
* ... 

This would allows to have approach linked of https://github.com/Slicer/Slicer/issues/6912#issuecomment-1482827047 all simplified. 

Also copied below for convenience:


> [...] following should be updated:
> 
> | Project | URL |
> |--|--|
> | SlicerExtension-VMTK | https://github.com/search?q=repo%3Avmtk%2FSlicerExtension-VMTK%20installExtensionFromServer&type=code |
> | SlicerMorph | https://github.com/SlicerMorph/SlicerMorph/blob/c86b3dd507ab0df9ad0fef721b1da1d69cc37d43/SegmentEndocranium/SegmentEndocranium.py#L51-L61 |
> | slicer-notebook | https://github.com/Slicer/SlicerDocker/blob/4fc3be96fd28cf97677650a76ac83f209efe834b/slicer-notebook/install.sh |
> | Slicer documentation | https://github.com/Punzo/Slicer/blob/08fe346009554f82e63e7d7aaaaec06c16f0a233/Docs/developer_guide/script_repository/gui.md?plain=1#L1137-L1147 |
> | DMRIInstall | https://github.com/Punzo/Slicer/blob/08fe346009554f82e63e7d7aaaaec06c16f0a233/Modules/Scripted/DMRIInstall/DMRIInstall.py#L111-L125 |
> | kaapana | https://github.com/kaapana/kaapana/blob/0a3b2e14f64510671c9c9dcb7ef73853002d8940/services/applications/slicer-workbench/docker/files/install-slicer-extension.py#L8 |

Reply:
I think we should just remove the old installation method. It would just confuse people.

Having a slicer-install pyrhon package would be great. Ideally, it should be usable for installing Slicer in Google colab.

Since slicerio package already can launch Slicer and call its methods (using web API), we may consider adding Slicer installation to this package. It could then download, install, and run Slicer without requiring the user to install or set up anything. Or slicerio could just use slicer-install for installing Slicer.

Reply:
This is now tracked in:
* https://github.com/Slicer/Slicer/issues/7150